interface/ceed.c

```diff
@@ -1182,4 +1182,10 @@ int CeedGetVersion(int *major, int *minor, int *patch, bool *release) {
   return 0;
 }
 
+int CeedGetScalarType(CeedScalarType *scalar_type) {
+  *scalar_type = CEED_SCALAR_TYPE;
+  return 0;
+}
```

Comment:
Do we want to commit to supporting this global interface, or might we do object-specific `CeedVectorGetScalarType(CeedVector, CeedScalarType *)`? I'm also fine if this is a stop-gap so we can get this PR to a mergeable state, and then we'll revisit. In any case, I'd hope we can avoid making heavy dependencies on scalar type being a compile-time constant.

Reply:
This was only added for the Julia interface, and that's the only place it's used at the moment.  My understanding of Will's comments was that it was necessary in order for Julia to know which definitions to use for things related to the scalar type (most importantly, `CeedScalar`), if the user provides a compiled binary of the library to use with the Julia interface.  An alternative was that the user would have to provide a location for the headers as well, but Will was not in favor of that.  We could maybe change its documentation (and even name?) to make it clear that it's specifically for the Julia interface, and not really meant to be used elsewhere?

Reply:
@pazner What if precision was per-object instead of per-library? (Perhaps every object would return the same thing now, but someday they could be different.) Would that serve your needs?

Reply:
It's a bit hard for me to say, because I don't completely understand what you are proposing. It could be more challenging if the type is not known globally at compile time, but rather per object at runtime, because in order for Julia to compile efficient code, it has to be "type stable", meaning that the types are inferable at compile time. It's possible that this would work with what you are describing, but I would need more information before being able to say for sure.

But I am also a bit confused: there are plenty of places in the C API that use `CeedScalar`. This type has to be known at compile time… or am I missing something? Would the proposal be to never pass `CeedScalar` by value and only use pointers instead (i.e. `CeedScalar` would be kind of like an opaque type)?

Reply:
This PR was meant to be a stepping stone toward a library in which scalar type could be chosen per-object similarly to how [BLIS supports fine-grained run-time selection of precision](https://github.com/flame/blis/blob/master/docs/MixedDatatypes.md). So for example, you could take a `CeedVector` in single precision and promote to double in restriction, or you could go the other way. User-provided QFunctions would have specified precision (or perhaps you'd set multiple variants if you have them available), but everything else would be run-time selectable. I think we have a strategy where users who don't want mixed precision will be able to basically ignore it by including a suitable header or using C11 `_Generic`, while those who want dynamically-selected mixed precision will also be able to do so.

Reply:
I see, that makes sense to me.

Since this PR is just a stepping stone, I think we can consider something like `CeedGetScalarType` to be a provisional solution to get the Julia interface working with this PR. It doesn't need to be included in the public API (it doesn't even have to be declared in the header file).

When it is clearer what the final C API for mixed precision will look like, we can try to come up with a design for the Julia interface that will work.